### PR TITLE
New assemble gls in sharp

### PR DESCRIPTION
**Pull request by Luckabarbeau**
Update the assembly in the gls sharp NS solver . Add the possibility to assemble a Poisson equation inside the particle.

### CODE REVIEW

**Review by blaisb — COMMENTED** [review]
I have a few comments. Also make sure all of your functions are well commented and documented.
If they are identical to the GLS functions, then don't recode them
:)

File: applications_tests/gls_sharp_navier_stokes_2d/couette2d_gls.prm [hunk 1]
```diff
@@ -90,7 +90,7 @@ subsection particles
 	set number of particles = 2
 	set stencil order = 6
 	set nb force evaluation = 1000
-
+	set assemble NS inside=true
```
[the remark this answers is not in the document]
Comment (Luckabarbeau) [reply applications_tests/gls_sharp_navier_stokes_2d/couette2d_gls.prm:93]: something more like : set use gls inside particles.  ?

File: source/solvers/navier_stokes_assemblers.cc [hunk 2]
```diff
@@ -825,3 +825,136 @@ GDNavierStokesAssemblerCore<dim>::assemble_rhs(
...
+            {
+              const auto &grad_phi_u_j = scratch_data.grad_phi_u[q][j];
+              const auto &grad_phi_p_j = scratch_data.grad_phi_p[q][j];
+
+              // Laplacian on the velocity terms
+              double local_matrix_ij =
+                h * h * scalar_product(grad_phi_u_j, grad_phi_u_i) * sdt;
+
+              // Laplacian on the pressure terms
+              local_matrix_ij += viscosity * h * h *
```
[the remark this answers is not in the document]
Comment (Luckabarbeau) [reply source/solvers/navier_stokes_assemblers.cc]: was rework

**Review by blaisb — APPROVED** [review]
I'll merge once the CI passes. Don't forget to update the wiki

**Pull request merged**